parts/k8s/kubernetesagentresourcesvmss.t

```diff
@@ -43,6 +43,10 @@
         "mode": "Manual"
       },
       "virtualMachineProfile": {
+        {{if .IsLowPriorityScaleSet}}
```

Comment:
Becuse we're simply assigning the `"priority"` value that is present in the `{{.Name}}ScaleSetPriority` variable, would it make more sense to remove this `{{if .IsLowPriorityScaleSet}}` clause? In other words, let's just make sure that a sane default value is always present for template variable expansion (either "Regular" or "Low").

The advantage of doing this is to remove an unnecessary `if` clause from the template (which already has a lot of conditional blocks and is rather hard to read!)

Reply:
Thanks Jack, I was in two minds on this and leant towards making the output template less verbose (whilst also reducing the number of parameters, and variables), despite setting the default values, as people who don't use or care about Low-priority VMSS aren't likely to encounter this in their templates (particularly because "Regular" is implicit). And also because it is currently preview. But happy to simplify if you prefer. Any thoughts?